pkg/component/shared/vali.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	hvpav1alpha1 "github.com/gardener/hvpa-controller/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/component/logging/vali"
+	"github.com/gardener/gardener/pkg/utils/images"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+// NewVali returns new Vali deployer
+func NewVali(
+	c client.Client,
+	namespace string,
+	imageVector imagevector.ImageVector,
+	secretsManager secretsmanager.Interface,
+	clusterType component.ClusterType,
+	replicas int32,
+	isLoggingEnabled, isShootNodeLoggingEnabled bool,
+	priorityClassName string,
+	storage *resource.Quantity,
+	ingressHost, ingressClass string,
+	authEnabled, hvpaEnabled bool,
+	maintenanceTimeWindow *hvpav1alpha1.MaintenanceTimeWindow,
```

Comment:
```suggestion
	c client.Client,
	namespace string,
	imageVector imagevector.ImageVector,
	secretsManager secretsmanager.Interface,
	clusterType component.ClusterType,
	replicas int32,
	isLoggingEnabled bool,
	priorityClassName string,
	storage *resource.Quantity,
        isShootNodeLoggingEnabled bool,
	ingressHost string,
	ingressClass string,
	authEnabled bool,
	hvpaEnabled bool,
	maintenanceTimeWindow *hvpav1alpha1.MaintenanceTimeWindow,
```

Nit: For better grouping/readability